### ISSUE TITLE
Fix assignment of `int` to `double`

### ISSUE THREAD

**Issue by perlun** [perlang-org/perlang#240]
The current operation fails:

```
Perlang Interactive REPL Console (0.1.0-dev.190, built from git commit 13539ef)
> var d: double = 1;
[line 1] Error at 'd': Cannot assign int to double variable
```

While this has a reasonable explanation at least for `long` to `double` (which can cause overflow because not all 64-bit integer are exactly representable in a `double`), it should be completely safe for `int` values.

----

We could also consider doing constant expression evaluation at compile-time, to make this possible for "safe" values where there is no precision loss. This is likely much more work though, but would be really nice to fix as well at some point.

**Comment by perlun** [comment]
This will have to wait for 0.2.0. It's not a _bug_ per se, but more like a "missing feature".

**Comment by perlun** [comment]
https://en.wikipedia.org/wiki/Double-precision_floating-point_format#Precision_limitations_on_integer_values states:

> * Integers from −2<sup>53</sup> to 2<sup>53</sup> (−9,007,199,254,740,992 to 9,007,199,254,740,992) can be exactly represented
> * Integers between 2<sup>53</sup> and 2<sup>54</sup> = 18,014,398,509,481,984 round to a multiple of 2 (even number)
> * Integers between 2<sup>54</sup> and 2<sup>55</sup> = 36,028,797,018,963,968 round to a multiple of 4

In other words, we could make it simple and say that "`int` to `double` conversion is always safe, but `long` to `double` is not". I'll try to give this a look to see if it's hard to fix.

**Comment by perlun** [comment]
Fixed as of #300, included in the 0.2.0 release. :partying_face: Closing.
